oi/FuncGen.cpp

```diff
@@ -274,6 +274,28 @@ void __attribute__((used, retain)) introspect_%2$016x(
       (boost::format(func) % type % std::hash<std::string>{}(type)).str());
 }
 
+void FuncGen::DefineTopLevelIntrospectNamed(std::string& code,
+                                            const std::string& type,
+                                            const std::string& linkageName) {
+  std::string typeHash =
+      (boost::format("%1$016x") % std::hash<std::string>{}(type)).str();
+
+  code += "/* RawType: ";
+  code += type;
+  code += " */\n";
+  code += "extern \"C\" IntrospectionResult ";
+  code += linkageName;
```

Comment:
This thing throws a warning because it's a complex C++ type being returned with C style linkage. However it seems to work fine, so I'm inclined not to care. The alternative could be to use a demangled name (which we get by demangling the mangled name) and hope it gets mangled the same again, but this seems more flaky.